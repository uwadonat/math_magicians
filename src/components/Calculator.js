import React, { useState } from 'react';
import './Calculator.css';
import calculate from '../logic/calculate';

function Calculator() {
  const [calcObject, setCalcObject] = useState({});

  function handleClick(e) {
    e.preventDefault();
    try {
      setCalcObject({ ...calcObject, ...calculate(calcObject, e.target.textContent) });
    } catch (error) {
      return calcObject;
    }
    return 0;
  }

  render() {
    const { next, total } = this.state;
    return (
      <div className="calculate-page">
        <h2>Let&apos;s do some Math</h2>
        <div className="calc-container">
          <div className="calc-display">
            {next || total || 0}
          </div>
          <button type="button" onClick={this.handleClick} className="gray-btn">AC</button>
          <button type="button" onClick={this.handleClick} className="gray-btn">+/-</button>
          <button type="button" onClick={this.handleClick} className="gray-btn">%</button>
          <button type="button" onClick={this.handleClick} className="orange-btn">รท</button>
          <button type="button" onClick={this.handleClick} className="gray-btn">7</button>
          <button type="button" onClick={this.handleClick} className="gray-btn">8</button>
          <button type="button" onClick={this.handleClick} className="gray-btn">9</button>
          <button type="button" onClick={this.handleClick} className="orange-btn">x</button>
          <button type="button" onClick={this.handleClick} className="gray-btn">4</button>
          <button type="button" onClick={this.handleClick} className="gray-btn">5</button>
          <button type="button" onClick={this.handleClick} className="gray-btn">6</button>
          <button type="button" onClick={this.handleClick} className="orange-btn">-</button>
          <button type="button" onClick={this.handleClick} className="gray-btn">1</button>
          <button type="button" onClick={this.handleClick} className="gray-btn">2</button>
          <button type="button" onClick={this.handleClick} className="gray-btn">3</button>
          <button type="button" onClick={this.handleClick} className="orange-btn">+</button>
          <button type="button" onClick={this.handleClick} className="gray-btn zero-btn">0</button>
          <button type="button" onClick={this.handleClick} className="gray-btn">.</button>
          <button type="button" onClick={this.handleClick} className="orange-btn">=</button>
        </div>
      </div>
    );
  }

}
export default Calculator;
