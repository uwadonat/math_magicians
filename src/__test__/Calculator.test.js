import React from 'react';
import renderer from 'react-test-renderer';

import Calculator from '../components/Calculator';

test('it should render correctly', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});

