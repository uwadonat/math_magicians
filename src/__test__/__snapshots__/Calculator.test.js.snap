// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render correctly 1`] = `
<div
  className="container page-border"
>
  <h2>
    Let's do some math!
  </h2>
  <div
    className="calc-container"
  >
    <div
      className="calc-display"
    >
      0
    </div>
    <button
      className="gray-btn"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="gray-btn"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="gray-btn"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="orange-btn"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="gray-btn"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="gray-btn"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="gray-btn"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="orange-btn"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="gray-btn"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="gray-btn"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="gray-btn"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="orange-btn"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="gray-btn"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="gray-btn"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="gray-btn"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="orange-btn"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="gray-btn"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="gray-btn"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="orange-btn"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
